### PR TITLE
Fix progress bar regression from dummy progress tracking for annotation

### LINKED ISSUE #6731: Crash when trying to run importcontent in command line.
### Observed behavior
A crash happens when running this management command in the Command Line/Terminal: 
* `kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5`

This sample content is for the Khan Academy (English).  Here's the user guide for reference: [Import Content from Internet](https://kolibri.readthedocs.io/en/latest/manage/command_line.html#import-content-channels-from-internet).

Here's the output:

```bash
user@ubuntu:~$ kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importcontent network 1ceff53605e55bef987d88e0908658c5
Traceback (most recent call last):
  File "/usr/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri==0.13.2b1', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 243, in invoke
    return super(KolibriDjangoCommand, self).invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 641, in manage
    execute_from_command_line(["kolibri manage"] + ctx.args)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 102, in handle
    return self.handle_async(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 439, in handle_async
    renderable_only=options["renderable_only"],
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 159, in download_content
    renderable_only=renderable_only,
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 257, in _transfer
    total=total_bytes_to_transfer + dummy_bytes_for_annotation
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 115, in start_progress
    total=total, level=level, update_callback=self._update_all_progress
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 38, in __init__
    self.progressbar = click.progressbar(length=total, width=0)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/termui.py", line 351, in progressbar
    width=width, color=color)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/_termui_impl.py", line 81, in __init__
    iterable = range_type(length)
TypeError: 'float' object cannot be interpreted as an integer
user@ubuntu:~$
```

### Expected behavior
No crash should happen and that the Channel content is imported successfully.

### Errors and logs

* [daemon.txt](https://github.com/learningequality/kolibri/files/4410316/daemon.txt)
* [kolibri.txt](https://github.com/learningequality/kolibri/files/4410317/kolibri.txt)

### Steps to reproduce
1. Install [Kolibri v0.13.2-beta1](https://github.com/learningequality/kolibri/releases/tag/v0.13.2-beta1) in Ubuntu 16.04.6 VM
1. Launch Terminal, then run `kolibri manage importchannel network 1ceff53605e55bef987d88e0908658c5` - ignore the warning
1. Run `kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5` - notice the crash.

### Context
 
* Kolibri version: [Kolibri v0.13.2-beta1](https://github.com/learningequality/kolibri/releases/tag/v0.13.2-beta1)
* Operating system: Ubuntu 16.04.6 VM (2GB RAM, 2CPUs, 64GB disk)
* Browser: Firefox v74.0

### Related Issue/s

* https://github.com/learningequality/kolibri/issues/6715#issuecomment-606696997

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
#6629 introduced non-integer progress totals to add dummy progress for the annotation task on large channels.
This resulted in a bug in progressbar updating, as it only handles integers #6731 
This PR fixes this by making sure the dummy progress is an integer, and giving a warning and coercing values being sent to the progress bars to integers.

### Reviewer guidance
Import a large channel from the command line in Python3. Verify that it works and does not error.

### References
Fixes #6731 

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by cpauya** [comment]
Tried the `.deb` in the build above and got these "logs" when I ran these:
* `kolibri manage importchannel network 1ceff53605e55bef987d88e0908658c5`
* `kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5`

```bash
user@ubuntu:~$ kolibri manage importchannel network 1ceff53605e55bef987d88e0908658c5
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importchannel network 1ceff53605e55bef987d88e0908658c5
INFO     Downloading data for channel id 1ceff53605e55bef987d88e0908658c5
WARNING  total argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
  [--------------------------------------------------------------------------------------------------------------------------------]    0%WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
  [#-------------------------------------------------------------------------------------------------------------------------------]    0%WARNING  increment argument should be an integer for progressbars to work
  [#-------------------------------------------------------------------------------------------------------------------------------]    1%WARNING  increment argument should be an integer for progressbars to work
...
... 1,700+ of lines
...
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
INFO     Importing ContentTag data
INFO     Importing ContentNode_has_prerequisite data
INFO     Importing ContentNode_related data
INFO     Importing ContentNode_tags data
INFO     Importing ContentNode data
INFO     Importing Language data
INFO     Importing File data
INFO     Importing LocalFile data
INFO     Importing AssessmentMetaData data
INFO     Importing ChannelMetadata data
INFO     Channel 1ceff53605e55bef987d88e0908658c5 successfully imported into the database
user@ubuntu:~$ 
```

Not sure if we should remove the warning logs coz it's not showing correctly in Ubuntu 16.04.6 VM.
Screenshot 2020-04-02 15 42 54


But no more crash yey! 😸

**Comment by rtibbles** [comment]
Interesting - so maybe it's just the total that needs to be an integer, and not the increments? Will double check and update the PR if so.

**Comment by rtibbles** [comment]
OK, yeah - click does not do any coercion that would cause an issue for the increment argument, so I'll remove that change, and fix the tests here.

**Comment by cpauya** [comment]
Thanks.  Will re-test this on the same Ubuntu 16.04.6 VM today.

I left VM running (with build from original changes) with the `kolibri manage importcontent...` command since yesterday.  This morning I got this error log: I don't know if this is relevant so I'm just posting it here:

```bash
user@ubuntu:~$ kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5
...
... I lost the results here due to the scrollback limit of the Terminal.
...
  [########################################################---------------------------------------------]   56%  10:40:30WARNING  increment argument should be an integer for progressbars to work
  [###############################################################################################################]  100%WARNING  increment argument should be an integer for progressbars to work
WARNING  total argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
WARNING  increment argument should be an integer for progressbars to work
...
... warnings repeated 5x
...
ERROR    An error occurred during content import: 520 Server Error:  for url: https://studio.learningequality.org/content/storage/6/1/61497dc0f7d71c1e23bdd9e7237d5baf.vtt
INFO     Setting availability to True of 58228 LocalFile objects based on passed in checksums
INFO     Setting availability of non-topic ContentNode objects based on LocalFile availability in 9 batches of 10000
INFO     Annotating ContentNode objects with children for 5 levels
INFO     Annotating ContentNode objects with children for level 5
INFO     Annotating ContentNode objects with children for level 4
INFO     Annotating ContentNode objects with children for level 3
INFO     Annotating ContentNode objects with children for level 2
INFO     Annotating ContentNode objects with children for level 1
WARNING  increment argument should be an integer for progressbars to work
  [#########################################################--------------------------------------------]   57%  10:24:19Traceback (most recent call last):
  File "/usr/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri==0.13.2b1.dev0+git.47.g47aaec40', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 243, in invoke
    return super(KolibriDjangoCommand, self).invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 641, in manage
    execute_from_command_line(["kolibri manage"] + ctx.args)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 118, in handle
    return self.handle_async(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 439, in handle_async
    renderable_only=options["renderable_only"],
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 159, in download_content
    renderable_only=renderable_only,
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 353, in _transfer
    raise exception
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 305, in _transfer
    f, filetransfer, overall_progress_update
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 409, in _start_file_transfer
    retry = retry_import(e, skip_404=True)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/import_export_content.py", line 159, in retry_import
    raise e
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 374, in _start_file_transfer
    with filetransfer, self.start_progress(
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/transfer.py", line 102, in __enter__
    self.start()
  File "/usr/lib/python3/dist-packages/kolibri/core/content/utils/transfer.py", line 162, in start
    self.response.raise_for_status()
  File "/usr/lib/python3/dist-packages/kolibri/dist/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 520 Server Error:  for url: https://studio.learningequality.org/content/storage/6/1/61497dc0f7d71c1e23bdd9e7237d5baf.vtt
user@ubuntu:~$ 
```

This could be the [520 Web Server Returned an Unknown Error](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#520) as per Wikipedia.  Here's another reference I found: [Community Tip - Fixing Error 520: Web server is returning an unknown error](https://community.cloudflare.com/t/community-tip-fixing-error-520-web-server-is-returning-an-unknown-error/44205).  A note, I was also downloading updates from Steam in a Windows machine and that stopped also when I checked this morning.

I didn't look at the code, but maybe we can add this error code in the exception handler to prevent crashes like this?  An alternative perhaps is a retry mechanism of some sort?

**Comment by cpauya** [comment]
I installed the latest build [0.13.2b1.dev0+git.48.g775989bf](https://buildkite.com/organizations/learningequality/pipelines/kolibri-python-package/builds/239/jobs/2a30f2d7-a55a-4e44-be30-7e8585695a52/artifacts/9756449c-4c12-4d96-924d-3839010b1753) and re-run Kolibri in the VM this morning to check and KA-Eng was indeed downloaded and can be used even with the HttpError 520 encountered above.  

Screenshot 2020-04-03 09 49 32

When I ran `kolibri manage importchannel...` the progress looks right.  But when I ran `kolibri manage importcontent...` the progress scrolls per item progress?  

Screenshot 2020-04-03 10 02 12

Ah, maybe the 0% is the total progress whereas the 100% is the per item progress?  Here's the output as of now: it's showing 4% total progress:
kolibri-v0 13 2-b1--terminal progress of importcontent

I'm running it in the background until it finishes. 😸

**Comment by cpauya** [comment]
Yey! Finished with `importcontent`!  

kolibri-v0 13 2-b1--importcontent finished

If the HttpError 520 above is a non-issue then we can merge this. 😸

**Comment by rtibbles** [comment]
I am surprised by the number of warnings - if a non-integer total was a problem, I don't understand why these didn't raise errors before. I think I'd like to fix up the instances where this warning is happening before we merge this.

**Comment by rtibbles** [comment]
I think my warning might be too aggressive, so I'm going to remove it.

**Comment by rtibbles** [comment]
The 520 is a known issue, so should be fine to merge anyway.

**Comment by cpauya** [comment]
Thanks!  Will test [the latest build](https://buildkite.com/learningequality/kolibri-python-package/builds/258) today and will merge if all good.

**Comment by cpauya** [comment]
Logging now looks good!  Merging this. 

kolibri-v0 13 2-b1--importcontent test 2020-04-04-sat-a

**Review by cpauya — APPROVED** [review]
Latest build works without crash as per last [test done](https://github.com/learningequality/kolibri/pull/6739#issuecomment-608977713) - merging.

**Pull request merged**
